userspace/falco/app/actions/init_inspectors.cpp

```diff
@@ -29,18 +29,41 @@ static void init_syscall_inspector(falco::app::state& s, std::shared_ptr<sinsp>
 {
 	inspector->set_buffer_format(s.options.event_buffer_format);
 
+	//
+	// Container engines
+	//
+
+	// Container engines configs via falco.yaml
+	inspector->set_container_engine_mask(s.config->m_container_engines_mask);
+	for (auto &p : s.config->m_container_engines_cri_socket_paths)
+	{
+		if (!p.empty())
+		{
+			inspector->add_cri_socket_path(p);
+			falco_logger::log(falco_logger::level::DEBUG, "Enabled container runtime socket at '" + p + "' via config file");
+		}
+	}
+	inspector->set_cri_async(!s.config->m_container_engines_disable_cri_async);
+
+	// Container engines configs via CLI args
 	// If required, set the CRI paths
 	for (auto &p : s.options.cri_socket_paths)
 	{
 		if (!p.empty())
 		{
 			inspector->add_cri_socket_path(p);
+			falco_logger::log(falco_logger::level::DEBUG, "Enabled container runtime socket at '" + p + "' via CLI args");
 		}
 	}
 
 	// Decide whether to do sync or async for CRI metadata fetch
 	inspector->set_cri_async(!s.options.disable_cri_async);
 
+	if(s.options.disable_cri_async || s.config->m_container_engines_disable_cri_async)
+	{
+		falco_logger::log(falco_logger::level::DEBUG, "Disabling async lookups for 'CRI'");
```

Comment:
Would be nice to extend the API and add a `is_cri_async` and a `get_cri_socket_paths` methods. If you also think it would be good I can do it during this dev cycle, but no rush.

Reply:
https://github.com/falcosecurity/libs/issues/2022 also added a label "good first issue" if someone wanted to get started helping us.